provider/src/test/kotlin/org/gradle/kotlin/dsl/provider/ScriptApiTest.kt

```diff
@@ -0,0 +1,165 @@
+package org.gradle.kotlin.dsl.provider
+
+import org.gradle.kotlin.dsl.KotlinBuildScript
+import org.gradle.kotlin.dsl.KotlinSettingsScript
+
+import org.gradle.api.Action
+
+import kotlin.reflect.KCallable
+import kotlin.reflect.KClass
+import kotlin.reflect.KFunction
+import kotlin.reflect.KMutableProperty
+import kotlin.reflect.KParameter
+import kotlin.reflect.KProperty
+import kotlin.reflect.KType
+import kotlin.reflect.KTypeProjection
+import kotlin.reflect.KVariance
+import kotlin.reflect.KVisibility
+import kotlin.reflect.full.createType
+import kotlin.reflect.full.declaredMembers
+import kotlin.reflect.full.valueParameters
+import kotlin.reflect.full.withNullability
+import kotlin.reflect.jvm.javaGetter
+import kotlin.reflect.jvm.jvmErasure
+
+import org.hamcrest.CoreMatchers.equalTo
+import org.junit.Assert.assertThat
+import org.junit.Test
+
+
+class ScriptApiTest {
+
+    @Test
+    fun `build script template implements script api`() =
+        assertThat(
```

Comment:
:o: I wonder if it wouldn't be clearer and require a little code to first convert the candidate members to some canonical representation (data class or string) and then assert something like `assertThat(candidateApiMembers, hasItems(apiMembers))` where `candidateApiMembers` and `apiMembers` are of type `List<<canonical representation>>`.

Reply:
I'd say that it looks tempting but the reflect api is already somehow canonical. There are a lot of cases to cover and that would probably be quite a lot of code.

The existing assertion outputs the missing members nicely when there are any.